### ISSUE TITLE
Support for external refs?

### ISSUE THREAD

**Issue by dradetsky** [cellular/oazapfts#21]
Do you have any plans to add support for external refs soon? That is, a definition of the form

```
MyObjectName:
  $ref: 'path/to/file.yaml#path/to/MyObjectName'
```

Background:

My company is making a lot of microservices. I'm trying to introduce schema-first design for these, and ultimately hoping to add utilities for simplifying this to our company typescript library. Naturally, I want to get all of the value I can out of the schema, like generating typescript types. However, I strongly suspect I'll ultimately need to pull common bits of the schema (like the types of error responses) into a library so they can be standard across many services.

I considered trying to add external ref support myself. I know that [openapi-backend](https://github.com/anttiviljami/openapi-backend) supports external refs, and they use the same parser as oazapfts, so I figure I could copy what they did. However:

0. I'm fairly busy, so I probably won't get to it very quickly.

1. I don't want to do it if you're already halfway done with it, because you'll probably do a better job.

**Comment by fgnass** [comment]
I think the basic idea would be to turn external refs into `import MyObjectName from "path/to/file.ts"`. Is it safe to assume that the relative paths of the API specs are the same as the ones of the generated TypeScript files? Otherwise we would need some form of mapping config.

**Comment by dradetsky** [comment]
@fgnass It sounds like instead of actually resolving external refs, you want to just recursively invoke the oazapfts cli on all the files named in the external refs, is that right?

If so, one issue with this is that you can't actually just invoke the cli on a file which is a valid external ref, but not an actual complete openapi definition. For example I can do

```
# file0.yaml
components:
  schemas:
    MyObj:
      $ref: 'file1.yaml#/MyObj'
---
# file1.yaml
MyObj:
  type: object
  ...
```

And this is valid openapi, but the oazapfts cli can't be invoked on file1 because it's not a valid openapi spec, just a valid fragment of one. Maybe this is possible to do by using just the various internal functions of oazapfts instead of the cli or main entry point, but I don't actually know the code yet.

The other point is that if you look at [openapi-backend](https://github.com/anttiviljami/openapi-backend/blob/master/src/backend.ts#L180) you'll see that it looks like they solve this with just one more call (`SwaggerParser.dereference`) to the parser library you both share. Maybe there's more to it than that, but I figure that at least you might be able to tell immediately that it wouldn't just work.

**Comment by dradetsky** [comment]
@fgnass i think i figured it out, pr incoming

**Comment by fgnass** [comment]
Fix landed in `oazapfts@3.0.0-rc4`
